lib/foreman_maintain/repository_manager/el.rb

```diff
@@ -0,0 +1,43 @@
+module ForemanMaintain::RepositoryManager
+  class El
+    include ForemanMaintain::Concerns::OsFacts
+    include ForemanMaintain::Concerns::SystemHelpers
+
+    def disable_repos(repo_ids)
+      if el7?
+        execute!("yum-config-manager --disable #{repo_ids.join(',')}")
+      else
+        execute!("dnf config-manager --set-disabled #{repo_ids.join(',')}")
+      end
+    end
+
+    def enable_repos(repo_ids)
+      if el7?
+        execute!("yum-config-manager --enable #{repo_ids.join(',')}")
+      else
+        execute!("dnf config-manager --set-enabled #{repo_ids.join(',')}")
+      end
+    end
+
+    def enabled_repos_hash
+      yum_cmd = 'yum repolist enabled -d 6 -e 0 2> /dev/null'
```

Comment:
Removed the grep and used ruby's grep instead.
Hmm, I am somewhat confused on if I should put `repolist` in package manager or in repository manager?

Reply:
To solve this confusion, I used package_manager to get the yum/dnf and then use that in cmd. Marking solved accordingly.